app/controllers/concerns/cache_concern.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module CacheConcern
+  extend ActiveSupport::Concern
+
+  def render_with_cache(**options)
+    raise ArgumentError, 'only JSON render calls are supported' unless options.key?(:json) || block_given?
+
+    key        = options.delete(:key) || [[params[:controller], params[:action]].join('/'), options[:json].respond_to?(:cache_key) ? options[:json].cache_key : nil, options[:fields].nil? ? nil : options[:fields].join(',')].compact.join(':')
```

Comment:
The cache key being dependent on the controller's action would mean that `StatusesController#show` and `StatusesController#activity` would have different cache keys, which isn't optimal. Not sure that would really be a problem in practice, though. What is `StatusesController#activity` used for anyway?

Reply:
That would be the correct behavior because those actions return different json. Activity is used for the resolveable id of activities.

Reply:
Do they really return different json?
Hm, also… that's not directly related, but `#activity` doesn't seem to require signatures even if `authorized_fetch_mode?` is true

Reply:
The show returns a Note object, the activity returns a Create or Announce object.

Reply:
Oh, right! Sorry about that.